x-pack/plugins/monitoring/server/telemetry_collection/get_logstash_stats.ts

```diff
@@ -324,37 +394,42 @@ export async function fetchLogstashState(
   ephemeralIds: string[],
   start: string,
   end: string,
-  { page = 0, ...options }: { page?: number } & LogstashProcessOptions
+  { page = 0, ...options }: { page?: number } & LogstashProcessOptions,
+  isSelfMonitoring: boolean = false
 ): Promise<void> {
+  const stateField = isSelfMonitoring ? 'logstash_state' : 'logstash.node.state';
+  const filterPath: string[] = [
+    `hits.hits._source.${stateField}.pipeline.batch_size`,
+    `hits.hits._source.${stateField}.pipeline.workers`,
+    `hits.hits._source.${stateField}.pipeline.representation.graph.vertices`,
+    `hits.hits._source.type`,
+  ];
+
   const params: estypes.SearchRequest = {
     index: INDEX_PATTERN_LOGSTASH,
     ignore_unavailable: true,
-    filter_path: [
-      'hits.hits._source.logstash_state.pipeline.batch_size',
-      'hits.hits._source.logstash_state.pipeline.workers',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices.config_name',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices.plugin_type',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices.meta.source.protocol',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices',
-      'hits.hits._source.type',
-    ],
+    filter_path: filterPath,
     body: {
       query: createQuery({
-        start,
-        end,
+        // intentionally not using start and end periods as we need recent node state info to fill plugin usages
+        // especially with metricbeat monitoring
         filters: [
-          { terms: { 'logstash_state.pipeline.ephemeral_id': ephemeralIds } },
+          { terms: { [`${stateField}.pipeline.ephemeral_id`]: ephemeralIds } },
           {
             bool: {
-              must: { term: { type: 'logstash_state' } },
+              should: [
+                { term: { type: 'logstash_state' } },
+                { term: { 'metricset.name': 'node' } },
+              ],
             },
           },
         ],
       }) as estypes.QueryDslQueryContainer,
-      from: page * HITS_SIZE,
-      collapse: { field: 'logstash_state.pipeline.ephemeral_id' },
+      collapse: {
+        field: `${stateField}.pipeline.ephemeral_id`,
+      },
       sort: [{ ['timestamp']: { order: 'desc', unmapped_type: 'long' } }],
-      size: HITS_SIZE,
+      size: 10, // we need recent
```

Comment:
review note: I don't think we need huge amount of list here, last 10 seems okay as one doc contains all pipeline definitions.